src/Program.ts

```diff
@@ -533,11 +536,50 @@ export class Program {
                         range: util.createRange(0, 0, 0, Number.MAX_VALUE)
                     });
                 }
+
+                //for every unvalidated file, validate it
+                if (!file.isValidated) {
+                    this.plugins.emit('beforeFileValidate', {
+                        program: this,
+                        file: file
+                    });
+
+                    //call file.validate() IF the file has that function defined
+                    file.validate?.();
```

Comment:
Why would that happen?

Reply:
I'm planning on eventually adding the ability for plugins to register their own file types. So this is safety for when a custom file doesn't implement a validate function.